meta-mbl-distro/recipes-core/mbl-initramfs-init/files/initramfs-init-script.sh

```diff
@@ -67,15 +73,19 @@ ROOTFS_PARTITION="$(findfs LABEL=$ROOTFS_LABEL)"
 
 # Switch from initramfs to rootfs
 mkdir -p /mnt/rootfs
-mount "$ROOTFS_PARTITION" /mnt/rootfs
+if [ "$(mount "$ROOTFS_PARTITION" /mnt/rootfs)" ]; then
+    # if we fail to mount, try to fix the partition and mount it again
+    e2fsck -p -c -f "$ROOTFS_PARTITION"
+    mount "$ROOTFS_PARTITION" /mnt/rootfs
+fi
 
 mount --move /dev /mnt/rootfs/dev
 mount --move /proc /mnt/rootfs/proc
 mount --move /sys /mnt/rootfs/sys
 mount --move "$LOG_MOUNT_POINT" "/mnt/rootfs${LOG_MOUNT_POINT}"
 cd /mnt/rootfs
 
-echo "Switching to $ROOTFS_PARTITION"
+echo "Switching to $ROOTFS_PARTITION" || true
```

Comment:
Oh, we have another one which cant be moved. Disregard my previous comment in that case.